nixos/modules/services/matrix/draupnir.nix

```
@@ -0,0 +1,306 @@
+              The path where Draupnir stores its data.
+
+              ::: {.note}
+                If you want to customize where this data is stored, use a bind mount.
+              :::
+            '';
+          };
+
+          pantalaimon = lib.mkOption {
+            readOnly = true;
+            description = ''
+              `pantalaimon` settings (enables E2E Encryption support).
+              This property is read-only, please configure `services.draupnir.pantalaimon` instead!
+            '';
+            default = {};
+            type = lib.types.submodule {
+              freeformType = format.type;
+              options = {
+                use = lib.mkOption {
+                  type = lib.types.bool;
+                  default = cfg.pantalaimon.enable;
+                  defaultText = "cfg.pantalaimon.enable";
+                  readOnly = true;
+                  description = ''
+                    Whether to use `pantalaimon` for E2E encryption. Enabled if `services.draupnir.pantalaimon.enable` is `true`.
+                  '';
+                };
+                username = lib.mkOption {
+                  type = lib.types.str;
+                  default = cfg.pantalaimon.username;
+                  defaultText = "cfg.pantalaimon.username";
+                  readOnly = true;
+                  description = ''
+                    Account name on the Matrix homeserver. Configured in `services.draupnir.pantalaimon.username`.
+                  '';
+                };
+              };
+            };
+          };
+          #endregion
+
+          #region Base settings
+          homeserverUrl = lib.mkOption {
+            type = lib.types.str;
+            defaultText = ''
+              URI to the pantalaimon instance, if configured. Else unset.
+            '';
+            description = ''
+              Base URL of the Matrix homeserver, that provides the Client-Server API.
+              If `services.draupnir.pantalaimon.enable` is `true`, this option will become read only. Configure `services.draupnir.pantalaimon.options.homeserver` instead in that case.
+              The listen address of `pantalaimon` will then become the `homeserverUrl` of `draupnir`.
+            '';
+          } // lib.optionalAttrs cfg.pantalaimon.enable {
+            default = "http://${config.services.pantalaimon-headless.instances."draupnir".listenAddress}:${toString config.services.pantalaimon-headless.instances."draupnir".listenPort}/";
+          };
+          #endregion
+
+          #region Common settings
+          autojoinOnlyIfManager = lib.mkOption {
+            type = lib.types.bool;
+            default = false;
+            description = ''
+              If `true`, the bot will only autojoin rooms if the user is a manager.
+            '';
+          };
+
+          automaticallyRedactForReasons = lib.mkOption {
+            type = lib.types.listOf lib.types.str;
+            default = [ ];
```

```suggestion
            default = [
              "spam"
              "advertising"
            ];
```
[Upstream default](https://github.com/the-draupnir-project/Draupnir/blob/v2.0.0-beta.4/config/default.yaml#L110).

---

Could probably set this to `null` instead, no? (and set defaultText to say itll use pustream defaults)
if im not mistaken the current default would unset the upstream defaulst, whereas omitting it in the generated config completely would use the defaults (as the config is merged into upstream defaults at runtime)

---

Yeah, sounds even better than having to track upstream defaults :+1: